In [1]:
import pandas as pd
import numpy as np

In [2]:
data = pd.read_csv("Sachin_ODI.csv")
data.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

In [3]:
data.shape

(360, 14)

# Observations:
- The dataset has 360 rows and 14 columns
- Each row represents one match data played by India and in total we have 360 matches data

# Calculate the Sample Space of the data?

In [4]:
# Exp : Whether Sachin has scored a century.
# All the matches Sachin has played.: 463
# The data is having 360 match records of Sachin playing for India in international matches

In [5]:
data.shape[0]

360

In [6]:
len(data)

360

# Event A: India winning the match

In [7]:
data["Won"].value_counts()

Won
True     184
False    176
Name: count, dtype: int64

In [9]:
data["Won"].value_counts(normalize = True) * 100

Won
True     51.111111
False    48.888889
Name: proportion, dtype: float64

### The sample space of event A is 184 from the above output

# Probability of Event A

In [10]:
184 / 360

0.5111111111111111

In [11]:
data["Won"].value_counts(normalize = True) 

Won
True     0.511111
False    0.488889
Name: proportion, dtype: float64

# Probability of A(complement)

In [12]:
1 - (184/360)

0.48888888888888893

In [13]:
176 / 360

0.4888888888888889

# Event B : Sachin scoring a century

In [14]:
data["century"].value_counts()

century
False    314
True      46
Name: count, dtype: int64

### The sample of Event B is 46

# Probability of Sachin scoring a century

In [15]:
46/360

0.12777777777777777

### If Sachin plays a ODI, there is only 12.8% chance that he will score a century

100 - 13 matches he will score a century.

# Probability of B (Complement)

In [18]:
1 - (46/360)

0.8722222222222222

### If Sachin Play an ODI, there is 87% chance that he will not score a century.

100 - 87

In [25]:
data[(data['runs']>=50) & (data['runs']< 100)]

runs  NotOut mins   bf  fours  sixes      sr  Inns           Opp  \
10     65       0   96   59      9      1  110.16     2   New Zealand   
13     70       0  137   91      8      0   76.92     2   West Indies   
14     90       0  124   84     14      1  107.14     2     Australia   
18     65       0   96   88      9      0   73.86     2     Sri Lanka   
25     57       0  104   71      6      0   80.28     2  South Africa   
..    ...     ...  ...  ...    ...    ...     ...   ...           ...   
333    69       0   91   63     10      1  109.52     1     Sri Lanka   
335    96       1  187  104     13      0   92.30     2     Sri Lanka   
347    53       0   93   68      7      0   77.94     2     Australia   
348    85       0  160  115     11      0   73.91     1      Pakistan   
359    52       0   93   48      5      1  108.33     2      Pakistan   

        Ground        Date        Winner    Won  century  
10      Nagpur  1995-11-26   New Zealand  False    False  
13     Gwalior  1996-02-21         India   True    False  
14      Mumbai  1996-02-27     Australia  False    False  
18     Kolkata  1996-03-13     Sri Lanka  False    False  
25     Sharjah  1996-04-19  South Africa  False    False  
..         ...         ...           ...    ...      ...  
333     Rajkot  2009-12-15         India   True    False  
335    Cuttack  2009-12-21         India   True    False  
347  Ahmedabad  2011-03-24         India   True    False  
348     Mohali  2011-03-30         India   True    False  
359      Dhaka  2012-03-18         India   True    False  

[75 rows x 14 columns]

In [26]:
75/360

0.20833333333333334

100 - 20 he will score a fifty

# Probability of India Winning the match given that  Sachin has scored a century